src/core/test.ts

```diff
@@ -0,0 +1,17 @@
+import { Evented, LeafletEvent } from "./Events";
+
+interface X extends LeafletEvent {
+  myProp: number
+}
+
+const e = new Evented<{
+  'test': X
+}>()
+
+e.on('test', (ev) => {
+  ev.myProp
+})
+
+e.on('jfdsksdfsldf', (e)=>{
+  
+})
```

Comment:
Remove this part or rename `jfdsksdfsldf``.

A real sample would be good to understand it better. Marker, Polygone or so (one where only the layer has the event and not the map)